functionapp.tf

```diff
@@ -0,0 +1,105 @@
+//Appserviceplan
+resource "azurerm_app_service_plan" "asp" {
+  location            = var.location
+  name                = format("%s-%s-%s", var.project, var.env, "asp")
+  resource_group_name = azurerm_resource_group.rg.name
+  kind                = "FunctionApp"
+  reserved            = false
+  sku {
+    size = var.aspskusize
+    tier = var.aspskutier
+  }
+  tags = module.ctags.common_tags
+}
+
+
+//Azure keyvault
+resource "azurerm_key_vault" "kv" {
+  location            = var.location
+  name                = format("%s%s", var.project, var.env)
+  resource_group_name = azurerm_resource_group.rg.name
+  tenant_id           = data.azurerm_client_config.current.tenant_id
+  sku_name            = "standard"
+  tags                = module.ctags.common_tags
```

Comment:
what about using rbac for accessing it? `enable_rbac_authorization = true`

and then using roles rather than access policies?